src/ol/render/canvas/Executor.js

```diff
@@ -280,6 +280,7 @@ class Executor {
     context.lineTo.apply(context, p1);
     if (fillInstruction) {
       this.alignFill_ = /** @type {boolean} */ (fillInstruction[2]);
+      context.fillStyle = /** @type {import("../../colorlike.js").ColorLike} */ (fillInstruction[1]);
```

Comment:
Without a context.save/restore, doesn't this affect subsequent polygon fills?

Reply:
It would if the the color isn't set each time, but `.fill_()` seems to rely on that as it is already doing save and restore for rotation https://github.com/openlayers/openlayers/blob/master/src/ol/render/canvas/Executor.js#L1038

Reply:
Well, since this seems to be an issue only when decluttering, I guess you are right. But then again, wouldn't it be anough to set the `fillStyle` here only when we're decluttering?